src/tests_exhaustive.c

```diff
@@ -327,184 +340,94 @@ void test_exhaustive_sign(const secp256k1_context *ctx, const secp256k1_ge *grou
-        for (r = 1; r < order; r++) {
-            for (msg = 1; msg < order; msg++) {
-                for (key = 1; key < order; key++) {
-                    secp256k1_ge nonconst_ge;
-                    secp256k1_ecdsa_recoverable_signature rsig;
-                    secp256k1_ecdsa_signature sig;
-                    secp256k1_pubkey pk;
-                    secp256k1_scalar sk_s, msg_s, r_s, s_s;
-                    secp256k1_scalar s_times_k_s, msg_plus_r_times_sk_s;
-                    int recid = 0;
-                    int k, should_verify;
-                    unsigned char msg32[32];
-
-                    secp256k1_scalar_set_int(&s_s, s);
-                    secp256k1_scalar_set_int(&r_s, r);
-                    secp256k1_scalar_set_int(&msg_s, msg);
-                    secp256k1_scalar_set_int(&sk_s, key);
-                    secp256k1_scalar_get_b32(msg32, &msg_s);
-
-                    /* Verify by hand */
-                    /* Run through every k value that gives us this r and check that *one* works.
-                     * Note there could be none, there could be multiple, ECDSA is weird. */
-                    should_verify = 0;
-                    for (k = 0; k < order; k++) {
-                        secp256k1_scalar check_x_s;
-                        r_from_k(&check_x_s, group, k);
-                        if (r_s == check_x_s) {
-                            secp256k1_scalar_set_int(&s_times_k_s, k);
-                            secp256k1_scalar_mul(&s_times_k_s, &s_times_k_s, &s_s);
-                            secp256k1_scalar_mul(&msg_plus_r_times_sk_s, &r_s, &sk_s);
-                            secp256k1_scalar_add(&msg_plus_r_times_sk_s, &msg_plus_r_times_sk_s, &msg_s);
-                            should_verify |= secp256k1_scalar_eq(&s_times_k_s, &msg_plus_r_times_sk_s);
-                        }
-                    }
-                    /* nb we have a "high s" rule */
-                    should_verify &= !secp256k1_scalar_is_high(&s_s);
-
-                    /* We would like to try recovering the pubkey and checking that it matches,
-                     * but pubkey recovery is impossible in the exhaustive tests (the reason
-                     * being that there are 12 nonzero r values, 12 nonzero points, and no
-                     * overlap between the sets, so there are no valid signatures). */
-
-                    /* Verify by converting to a standard signature and calling verify */
-                    secp256k1_ecdsa_recoverable_signature_save(&rsig, &r_s, &s_s, recid);
-                    secp256k1_ecdsa_recoverable_signature_convert(ctx, &sig, &rsig);
-                    memcpy(&nonconst_ge, &group[sk_s], sizeof(nonconst_ge));
-                    secp256k1_pubkey_save(&pk, &nonconst_ge);
-                    CHECK(should_verify ==
-                          secp256k1_ecdsa_verify(ctx, &sig, msg32, &pk));
-                }
-            }
+    printf("test count = %i\n", count);
```

Comment:
This is where valgrind on MacOS complains about uninit data.

This is spurious, see https://stackoverflow.com/questions/29312335/valgrind-on-os-x-yosemite-giving-bogus-errors ... @elichai Do you think we can update to MacOS 10.13 ? According to homebrew, valgrind supports this now.

Reply:
Ah no, we're at 10.13 already... Upstream bug: https://bugs.kde.org/show_bug.cgi?id=399584 There they mention https://github.com/LouisBrunner/valgrind-macos as an alternative tap. Do you think it's worth trying?

Reply:
We can try, idk how much will we trust it though (if someone is bored enough, he can go over their patches hehe https://github.com/LouisBrunner/valgrind-macos/commits/patches)

Reply:
Anyone have a clue why the `tests` binary doesn't have the exact same problem?

Reply:
Seems that fixed it.